File: civitasJAVA/src/civitas/SorpresaJugadorEspeculador.java
package civitas;

import java.util.ArrayList;

/**
 * @author Jesús Navarro Merino
 * @author José Alberto Gómez García
 */

public class SorpresaJugadorEspeculador extends Sorpresa{

    private int valor;

    public SorpresaJugadorEspeculador(String texto,int _valor) {
        super(texto);
        valor=_valor;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
         if(jugadorCorrecto(actual,todos)){
           informe(actual,todos);
           todos.set(actual, new JugadorEspeculador(todos.get(actual),valor));
        }
    }
    
    @Override
    protected void informe (int actual, ArrayList<Jugador> todos){
        super.informe(actual, todos);
    }
    
} //Fin de clase SorpresaJugadorEspeculador

File: civitasRUBY/lib/sorpresa_jugador_especulador.rb
# encoding:utf-8

require_relative 'jugador'
require_relative 'jugador_especulador'


module Civitas
  class SorpresaJugadorEspeculador < Sorpresa
    
    attr_reader :valor, :texto
    
    def initialize(texto,_valor)
      super(texto)
      @valor=_valor      
    end
    
    def self.constructorSorpresaJugadorEspeculador(texto,valor)
      new(texto,valor)
    end
    
    def aplicarAJugador(actual,todos)
        if(jugadorCorrecto(actual,todos))
          informe(actual,todos)
          todos[actual]=JugadorEspeculador.nuevoEspeculador(todos[actual],@valor)
        end
    end
    
    def informe(actual,todos)
      super(actual,todos)
    end
    
    public_class_method :new
    
  end
end
